Documentation updates for Toolchain changes

SenRamakri opened the pull request:
Documentation updates for Toolchain changes planned for 5.12 release

SenRamakri committed:
Toolchain documentation updates

SenRamakri commented:
Please review - But DO NOT MERGE this until the compiler changes are done. We will update the comments in the PR once we are ready to merge this after tool chain work is done. I took a shortcut as of now to capture both IAR and ARMC changes in one PR. Its possible that one might happen but other might not, in that case we will update this PR accordingly. This covers - IOTCORE-825 and IOTCORE-709.

deepikabhavnani commented:
Iar doc update PR was created long back - https://github.com/ARMmbed/mbed-os-5-docs/pull/885

In case one is updated can use that

SenRamakri committed:
Update target configuration information for ARMC6/ARMC5

SenRamakri committed:
Update tools intro section to capture new requirements for ARMC6 tranisition

SenRamakri commented:
@AnotherButler @kegilbert - I added more changes to this PR as we changed our requirements around ARMC6 transition and that caused more documentation changes. Please review.

Commit:
Edit tools_intro.md

Make minor edits to include articles for consistency.

Commit:
Update mbed_targets.md

Edit file for minor formatting.

Commit:
Edit mbed_targets.md

Edit file for consistent capitalization and formatting.

AnotherButler commented:
Does this still have a code dependency?

AnotherButler commented:
@SenRamakri Is this ready to merge, or are we waiting on a code dependency?

SenRamakri commented:
> @SenRamakri Is this ready to merge, or are we waiting on a code dependency?

@AnotherButler - Unfortunately we need to wait, as IAR8 just merged today and we have to start on ARMC6 tomorrow. Sorry that we have to keep you waiting on this.

AnotherButler commented:
Could you please add a link here to the code PR?

SenRamakri commented:
@AnotherButler - Yes, actually we don't have a specific PR, we need the branch feature-armc6 merged before we can merge this documentation change. In fact, the merge PR is still not generated and Im waiting on this PR now - https://github.com/ARMmbed/mbed-os/pull/9860, sorry for the convoluted info.

bulislaw commented on docs/reference/configuration/mbed_targets.md:
```diff
@@ -192,7 +192,13 @@ Enabling `is_disk_virtual` adds delay after flashing firmware binary to make sur
-The `supported_toolchains` property is the list of toolchains that support a target. The allowed values for `supported_toolchains` are `ARM`, `uARM`, `ARMC6`, `GCC_ARM` and `IAR`.
+The `supported_toolchains` property is the list of toolchains that support a target. The allowed values for `supported_toolchains` are `ARM`, `uARM`, `ARMC5`, `ARMC6`, `GCC_ARM` and `IAR`.
+
+If you specify `ARMC5` in `supported_toolchains`, it means the corresponding target supports `Arm Compiler 5.06 update 6`.
```
We should be explicit that any new targets should support ARMC6 not 5.

SenRamakri replied:
@bulislaw - Please see the updates.

deepikabhavnani commented:
Needs a rebase to resolve conflicts

SenRamakri committed:
Added section mentioning ARMC6 as the default ARM toolchains

SenRamakri commented:
Rebased

Commit:
Edit addition to tools_intro.md

Edit note, mostly for branding and brevity.

Commit:
Edit addition to mbed_targets.md

Edit note, mostly for branding and active voice.

AnotherButler commented:
@bulislaw Have all of your concerns been addressed?

0xc0170 commented:
Master received ARMC6 update. We are close to 5.12rc1, this documentation should be part of it.

SenRamakri commented:
@AnotherButler - ARMC6/IAR8 changes are merged to master. I think we are good to merge these changes now.

AnotherButler commented:
@bulislaw I'm going to merge this later today. If you have further concerns, please create a new PR.
